recipes/boost/all/conanfile.py

```diff
@@ -1292,15 +1295,17 @@ def _toolset_tag(self):
             "apple-clang": "",
             "msvc": "vc",
             "Visual Studio": "vc",
+            "msvc": "vc",
         }.get(str(self.settings.compiler), str(self.settings.compiler))
         if (self.settings.compiler, self.settings.os) == ("gcc", "Windows"):
             compiler = "mgw"
         os_ = ""
         if self.settings.os == "Macos":
             os_ = "darwin"
-        toolset_version = str(tools.Version(self.settings.compiler.version).major)
-        if str(self.settings.compiler) in ("msvc", "Visual Studio"):
+        if self._is_msvc:
             toolset_version = self._toolset_version.replace(".", "")
```

Comment:
`_toolset_version()` relies on `tools.msvs_toolset()` which doesn't support `compiler=msvc`, but I don't see an equivalent function in `conan.tools.microsoft`.

/cc @memsharded @lasote